src/wh_server_nvm.c

```diff
@@ -262,20 +262,46 @@ int wh_Server_HandleNvmRequest(whServerContext* server,
             wh_MessageNvm_TranslateAddObjectDma32Request(magic,
                     (whMessageNvm_AddObjectDma32Request*)req_packet, &req);
 
-            /* TODO: Add hostaddr conversion/checking */
-            metadata =(void*)((intptr_t)req.metadata_hostaddr);
-            data = (void*)((intptr_t)req.data_hostaddr);
-            /* TODO: Add data_len checking */
+            /* perform platform-specific host address processing */
+            resp.rc = wh_Server_DmaProcessClientAddress32(
+                server, req.metadata_hostaddr, &metadata, sizeof(whNvmMetadata),
+                WH_DMA_OPER_CLIENT_READ_PRE, (whServerDmaFlags){0});
+            if (resp.rc != WH_ERROR_OK) {
+                goto transRespAddObjDma32;
```

Comment:
Yeah, not loving the goto.  I'd sooner refactor to add separate DMA handling functions to keep this function reasonably small.